src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,48 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import mate.academy.exception.NoUserInCacheException;
+import mate.academy.exception.RequestProcessingException;
 
 public class AsyncRequestProcessor {
+    private static Map<String, UserData> cache = new ConcurrentHashMap<>();
     private final Executor executor;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(
+                () -> findUserInCache(userId), executor).exceptionally(ex -> {
+                    try {
+                        UserData userData = emulateDataBaseAccess(userId);
+                        CompletableFuture.runAsync(() -> cacheFetchedData(userId, userData));
+                        return userData;
+                    } catch (InterruptedException e) {
+                        throw new RequestProcessingException("Failed to process "
+                                + "the request for userId=" + userId, e);
+                    }
+                });
+    }
+
+    private UserData findUserInCache(String userId) {
+        if (cache.containsKey(userId)) {
+            return cache.get(userId);
+        } else {
+            throw new NoUserInCacheException();
+        }
+    }
+
+    private void cacheFetchedData(String userId, UserData userData) {
+        cache.put(userId, userData);
+    }
+
+    private UserData emulateDataBaseAccess(String userId) throws InterruptedException {
+        Thread.sleep(1000);
+        return new UserData(userId, "Some %s user data".formatted(userId));
```

Comment:
Simulating database access with a fixed delay may not accurately represent the behavior of a real database and can lead to misleading conclusions during testing. Consider using a more sophisticated stub or mocking framework for testing database interactions.